nexus/db-model/src/dataset_kind.rs

```diff
@@ -41,6 +44,9 @@ impl From<internal::shared::DatasetKind> for DatasetKind {
             internal::shared::DatasetKind::InternalDns => {
                 DatasetKind::InternalDns
             }
+            internal::shared::DatasetKind::ZoneRoot => DatasetKind::ZoneRoot,
+            internal::shared::DatasetKind::Zone { .. } => DatasetKind::Zone,
```

Comment:
Could you add a comment here explaining what happens to the zone name? Looks like it's stored as another column in the dataset table.

Reply:
Sure thing, added!